src/test/codegen/static-relocation-model.rs

```diff
@@ -0,0 +1,25 @@
+// Verify linkage of external symbols in the static relocation model.
+//
+// compile-flags: -O -C relocation-model=static
+// aux-build: extern_decl.rs
+
+#![crate_type = "rlib"]
+
+extern crate extern_decl;
+
+// The `extern_decl` definitions are imported from a statically linked rust
+// crate, thus they are expected to be marked `dso_local` without `dllimport`.
+//
+// The `access_extern()` symbol is from this compilation unit, thus we expect
+// it to be marked `dso_local` as well, given the static relocation model.
+//
+// CHECK: @extern_static = external dso_local local_unnamed_addr global i8
+// CHECK: define dso_local i8 @access_extern() {{.*}}
+// CHECK: declare dso_local i8 @extern_fn() {{.*}}
+
+#[no_mangle]
+pub fn access_extern() -> u8 {
+    unsafe {
+        extern_decl::extern_fn() + extern_decl::extern_static
+    }
+}
```

Comment:
I suspect that this test will fail on at least some targets on CI, and you'll have to use some per-target checking like in `src\test\codegen\abi-efiapi.rs`, or limit it to uefi.